### PR TITLE
Kraken: Refacto isochrone

### PR DESCRIPTION
**Pull request by AurelienLP**
Readable commit by commit!

In kraken, `make_isochrone_common` was used for graphical_isochrone and heat_maps but not for isochrone...

It changes the behavior of graphical_isochrone and heat_maps when no stop_points were found around the center. We now returns a no solution response rather than a `no_origin_nor_destination` which is used when the request does not have a `origins` or `destinations`.

This will help with isochrone_distributed.

TL;DR: 3 APIs share the same code now.

### CODE REVIEW

File: source/routing/raptor_api.cpp [hunk 1]
```diff
@@ -1410,41 +1410,86 @@ void make_response(navitia::PbCreator& pb_creator,
...
+          bound(bound),
+          datetime(datetime) {}
+};
+
+static const boost::optional<IsochroneCommon> make_isochrone_common(
+    RAPTOR& raptor,
+    const type::EntryPoint& center,
+    const uint64_t departure_datetime,
+    const double max_duration,
+    uint32_t max_transfers,
```
Comment (benoit-bst) [inline source/routing/raptor_api.cpp]:
```suggestion
    const uint32_t max_transfers,
```
? 
Dunno for RAPTOR& raptor and georef::StreetNetwork& worker

**Review by benoit-bst — APPROVED** [review]
if there a **Utest** or something like that, into the code, to prove the proper functioning ? 
I image it already exists, probably in a global way.

File: source/routing/raptor_api.cpp [hunk 2]
```diff
@@ -1459,12 +1459,7 @@ static const boost::optional<IsochroneCommon> make_isochrone_common(
     if (!departures) {
         pb_creator.fill_pb_error(pbnavitia::Error::unknown_object, "The entry point: " + center.uri + " is not valid");
         return boost::optional<IsochroneCommon>{};
-    } else if (departures->empty()) {
```
Comment (pbougue) [inline source/routing/raptor_api.cpp]: :+1: nice!

**Review by pbougue — APPROVED** [review]

File: source/routing/raptor_api.cpp (hunk 1 above)
Comment (AurelienLP) [reply source/routing/raptor_api.cpp]: Can't for RAPTOR& raptor and georef::StreetNetwork& worker :cry:

**Comment by AurelienLP** [comment]
> if there a **Utest** or something like that, into the code, to prove the proper functioning ?
> I image it already exists, probably in a global way.

Yes, some Utests already exist, that is how I found the different behavior between APIs.

**Review by benoit-bst — APPROVED** [review]
